### DISCUSSION TITLE
安卓手机端能用串口吗？我的不能选择串口

### DISCUSSION THREAD

**Discussion by Loser9527** [wh201906/SerialTest#11] [Q&A] [answered]
手机型号是小米10，其它串口工具能看到我的OTG串口，但是再这个APP里不能选择串口
![b9c327eacd2a9bd4713089c68aeeb84](https://user-images.githubusercontent.com/37859319/186389986-fe5d856a-2c56-448e-915b-52342521df6b.jpg)
![ab366f6497c351d8d7946b1de47c510](https://user-images.githubusercontent.com/37859319/186390017-02c65c17-e135-48f6-89ac-cba7bcc6dbc0.jpg)

**Comment by wh201906** [comment, marked as answer]
目前Android端暂时不支持串口，替代方案是装一个[UART转TCP](https://play.google.com/store/apps/details?id=com.hardcodedjoy.tcpuart)的软件，然后在SerialTest里面用TCP连接
Android端串口要等一段时间才会做出来

**Reply by Loser9527** [reply]
好的，期待中，太感谢你们了，开源这么优秀的软件！

**Reply by wh201906** [reply]
感谢你的支持！
我把我平常用的UART转TCP的软件放在楼下了，你可以试一下

**Comment by wh201906** [comment]
UART转TCP软件
com.hardcodedjoy.tcpuart
https://play.google.com/store/apps/details?id=com.hardcodedjoy.tcpuart
直接下载：
[com.hardcodedjoy.tcpuart_8_apps.evozi.com.zip](https://github.com/wh201906/SerialTest/files/9415123/com.hardcodedjoy.tcpuart_8_apps.evozi.com.zip)

**Comment by wh201906** [comment]
SerialTest supports hardware serial port in `/dev/`. You need to run `chmod 777 <serial port device>` with root permission to make the serial port accessible to SerialTest.
SerialTest doesn't support USB to serial adapters like FT232, CH340, CP2102 and other USB-CDC devices natively. However, you can still use them with SerialTest, if you have a UART to TCP app to convert the serial connection to TCP connection. I got this idea from [this webpage](https://github.com/RfidResearchGroup/proxmark3/blob/master/doc/termux_notes.md#tcp-bridge-method) and it does work.

You can use this app for converting UART connection to TCP connection.
com.hardcodedjoy.tcpuart
https://play.google.com/store/apps/details?id=com.hardcodedjoy.tcpuart
***
SerialTest支持`/dev/`目录下的的硬件串口。但需要运行`chmod 777 <串口设备地址>`(需要root权限)来让SerialTest能够访问它。
SerialTest本身不支持各类USB串口设备(FT232、CH340、CP2102等，以及USB-CDC设备)。但是，你可以通过UART转TCP的APP来把串口转为TCP连接，从而通过SerialTest来访问。这一方法源于[Proxmark3的文档](https://github.com/RfidResearchGroup/proxmark3/blob/master/doc/termux_notes.md#tcp-bridge-method)，经测试能够正常工作。
UART转TCP软件请看上一帖
